src/disk_interface.h

```diff
@@ -51,9 +51,11 @@ struct DiskInterface: public FileReader {
   virtual bool MakeDir(const std::string& path) = 0;
 
   /// Create a file, with the specified name and contents
+  /// If \a crlf_on_windows is true, \n will be converted to \r\n (only on
+  /// Windows builds of Ninja).
   /// Returns true on success, false on failure
-  virtual bool WriteFile(const std::string& path,
-                         const std::string& contents) = 0;
+  virtual bool WriteFile(const std::string& path, const std::string& contents,
+                         bool crlf_on_windows) = 0;
```

Comment:
I suggest a different way to achieve this that will be far less instrusive (e.g. will keep most unit-tests unmodified), as well as more readable (boolean arguments can be hard to read at call sites):

1) Do not touch `WriteFile()` here, but instead add a `virtual WriteTextFile(...);` method with a default implementation that just calls `WriteFile()`.

2) Override it in `ReadDiskInterface::WriteTextFile()` on _WIN32 only, to perform the write the new option.

3) Modify the call site that writes the response file to use `WriteTextFile()`.

This also needs a new unit-test to verity that WriteTextFile() works as expected (with different expected results on Posix and Windows). Wdyt?